pkg/applicationserver/io/pubsub/provider/connection.go

```diff
@@ -96,14 +85,41 @@ type Connection struct {
 
 // Shutdown shuts down the topics, subscriptions and the connections if required.
 func (c *Connection) Shutdown(ctx context.Context) error {
-	for _, s := range []Shutdowner{
+	return shutdown(ctx,
 		&c.Topics,
 		&c.Subscriptions,
 		c.ProviderConnection,
-	} {
+	)
+}
+
+var errShutdownFailed = errors.DefineInternal("shutdown_failed", "failed to shutdown")
```

Comment:
No "failed" in var name, error name, nor description. It's an error so `errShutdown` is clear enough (with name `shutdown` and description `shutdown`)